File: 2021/day01/a.cpp
#include <iostream>

using namespace std;

int main()
{
	int larger = 0;
	int current = 0;
	int previous = 0;

	cin >> previous;

	while (cin >> current)
	{
		if (current > previous)
			larger++;
		previous = current;
	}

	cout << "larger count: " << larger << endl;
}

File: 2021/day01/a.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();
    let mut larger = 0u32;
    let mut prev = u32::MAX;

    while let Some(line) = lines.next() {
        let input = line.unwrap();
        match input.parse::<u32>() {
            Ok(cur) => {
                if cur > prev {
                    larger += 1;
                }
                prev = cur;
            },
            Err(..) => {},
        }
    }

    println!("larger count: {}", larger);
}
